topcoffea/modules/datacard_tools.py

```diff
@@ -796,10 +803,10 @@ def analyze(self,km_dist,ch,selected_wcs):
                     if self.use_real_data:
                         if len(data_sm) != 1:
                             raise RuntimeError("obs data has unexpected number of sparse bins")
-                        elif sum(data_obs) != 0:
+                        elif sum(data_obs[0]) != 0:
                             raise RuntimeError("filling obs data more than once!")
                         for sp_key,arr in data_sm.items():
-                            data_obs += arr
+                            data_obs[0] += arr[0]
```

Comment:
for some reason my "above change" didn't get saved and posted. What I was referring to was the following line that declares the `data_obs` array:
```python
data_obs = [np.zeros(ch_hist._dense_shape[0] - 1), np.zeros(ch_hist._dense_shape[0] - 1)]
```
This can be changed to
```python
data_obs = np.zeros((2,ch_hist._dense_shape[0] - 1))
```
Which keeps `data_obs` as an `np.array` and allows for the more compact array addition syntax I mentioned here and elsewhere.